Add vault logs documentation and test env (#5365)
DataDog/integrations-core: vault/README.md

| `<INIT_CONFIG>`      | blank or `{}`                            |
| `<INSTANCE_CONFIG>`  | `{"api_url": "http://%%host%%:8200/v1"}` |

#### Log Collection

**Available for Agent >6.0**

1. Collecting logs is disabled by default in the Datadog Agent, enable it in your `datadog.yaml` file:

```
logs_enabled: true
```

2. Configure Vault to enable audit and server logs.

>     * Audit logs must enabled by a privileged user with the appropriate policies, [see more][11].

Reviewer: Typo, and copy suggestion:

```suggestion
    * Audit logs must be enabled by a privileged user with the appropriate policies. See [Enabling audit devices][11] for more information.
```

        ```
        $ vault audit enable file file_path=/vault/vault-audit.log
        ```

>     * [Server logs][12] are not written to file by default. You can configure static server logs in the [start up script][13].

Reviewer: Copy suggestion:

```suggestion
    * Make sure that [server logs][12] are written to a file. You can configure static server logs in the [Vault systemd startup script][13].
```

        The following script is outputting the logs to `/var/log/vault.log`.
        ```
        ...
        [Service]
        ...
        ExecStart=/bin/sh -c '/home/vagrant/bin/vault server -config=/home/vagrant/vault_nano/config/vault -log-level="trace" > /var/log/vault.log
        ...
        ```
 
3. Add this configuration block to your `vault.d/conf.yaml` file to start collecting your Vault logs:
    ````yaml
    logs:
    - type: file
        path: /vault/vault-audit.log
        source: vault

>         service: vault

Reviewer 2: ```suggestion
        service: <SERVICE_NAME>
```

Reviewer 2: Indentation to fix as well

    - type: file
        path: /var/log/vault.log

>         source: vault

Reviewer: Should this block have `service: vault` as well?

Reviewer 2: Let's add `service: <SERVICE_NAME>`
